src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -191,7 +197,7 @@
~        _isExternalErrorDiagnosticUpdateSourceSubscribedToSolutionBuildEvents = true;
~    }
~
~    public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asyncServiceProvider)
```

Comment:
>var openFileTracker = await OpenFileTracker.CreateAsync(this, ProjectSystemProjectFactory, asyncServiceProvider).ConfigureAwait(true); [](http://example.com/codeflow?start=8&length=134)

Does the SwitchToMainThreadAsync call above still need to always yield since the amount of work being done is much less now?

Reply:
Oh, maybe getting the telemetry session is still potentially costly

Reply:
I'm ahead of you! https://github.com/dotnet/roslyn/pull/79282

Reply:
I'm leaving it, since once this is in, I'm doing a follow-up to delete the initialize method entirely since the only remaining work would be the UIContext stuff which doesn't need the UI thread anymore either.